src/status_im/hardwallet/fx.cljs

```diff
@@ -95,3 +98,32 @@
  :send-transaction-with-signature
  (fn [{:keys [transaction signature on-completed]}]
    (status/send-transaction-with-signature transaction signature on-completed)))
+
+(re-frame/reg-fx
+ :hardwallet/persist-pairings
+ (fn [pairings]
+   (.. js-dependencies/async-storage
+       (setItem "status-keycard-pairings" (types/clj->json pairings)))))
+
+(re-frame/reg-fx
+ :hardwallet/retrieve-pairings
+ (fn []
+   (when platform/android?
+     (.. js-dependencies/async-storage
+         (getItem "status-keycard-pairings")
+         (then #(re-frame/dispatch [:hardwallet.callback/on-retrieve-pairings-success
+                                    (types/json->clj %)]))))))
+
+(re-frame/reg-fx
+ :hardwallet/listen-to-hardware-back-button
+ (fn []
+   (re-frame/dispatch [:hardwallet/add-listener-to-hardware-back-button
```

Comment:
why is this one a dispatch?

Reply:
it dispatches to assoc listener to db

Reply:
ok are you sure this shouldn't be done in the screen will-mount and will-unmount method then? it doesn't seem right to do this through re-frame

Reply:
It could be done from screen will-mount event. But it's easier to make it as effect `listen-to-hardware-back-button` and use it in `fx/merge` as that effect should happen under different conditions and it's not dependent on particular screen to be loaded. It also could be tested once tests would be added.

Reply:
ok good point please add that explanation in the code